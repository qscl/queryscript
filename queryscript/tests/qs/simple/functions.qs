import * from inference;

select min(id) from users;
select max(id) from users;
select count(id) from users;
select count(*) from users;
select sum(id) from users;
select avg(id) from users;
select array_agg(id) from users;

select concat(id::text, 'a') from users;
select concat(id::text, 'a', 'b') from users;
select concat(id::text, 'a', 'b', 'c') from users;

select __native_identity(id) from users;
