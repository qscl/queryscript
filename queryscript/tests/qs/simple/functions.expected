{
    "compile_errors": [],
    "decls": {
        "let events": External<[{
        	description Utf8,
        	ts Utf8,
        	user_id Int64,
        }]>,
        "let users": External<[{
        	active Boolean,
        	id Int64,
        	name Utf8,
        	org_id Int64,
        }]>,
    },
    "queries": [
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "min(id)",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| min(id) |\n|---------|\n| 1       |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "max(id)",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| max(id) |\n|---------|\n| 2       |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "count(id)",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| count(id) |\n|-----------|\n| 2         |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "count(*)",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| count(*) |\n|----------|\n| 2        |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "sum(id)",
                                type_: Atom(
                                    Decimal128(
                                        38,
                                        0,
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| sum(id) |\n|---------|\n| 3       |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "avg(id)",
                                type_: Atom(
                                    Float64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| avg(id) |\n|---------|\n| 1.5     |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "ARRAY_AGG(id)",
                                type_: List(
                                    Atom(
                                        Int64,
                                    ),
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| ARRAY_AGG(id) |\n|---------------|\n| [1, 2]        |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "concat(CAST(id AS TEXT), 'a')",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| concat(CAST(id AS TEXT), 'a') |\n|-------------------------------|\n| 1a                            |\n| 2a                            |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "concat(CAST(id AS TEXT), 'a', 'b')",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| concat(CAST(id AS TEXT), 'a', 'b') |\n|------------------------------------|\n| 1ab                                |\n| 2ab                                |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "concat(CAST(id AS TEXT), 'a', 'b', 'c')",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| concat(CAST(id AS TEXT), 'a', 'b', 'c') |\n|-----------------------------------------|\n| 1abc                                    |\n| 2abc                                    |",
            },
        ),
        Err(
            Unimplemented {
                what: "Function parameters",
                backtrace: None,
            },
        ),
    ],
}